netcfgbu/config.py

```diff
@@ -32,12 +37,24 @@ def validation_errors(filepath, errors):
     return "\n".join(as_human)
 
 
-def load(*, filepath=None, fileio=None) -> AppConfig:
-    app_cfg = dict()
+def load(*, filepath: Optional[str] = None, fileio=None) -> AppConfig:
+    """Load and validate the application configuration from a TOML file.
+
+    Args:
+        filepath: Optional path to the configuration file.
+        fileio: Optional file object to read the configuration from.
+
+    Returns:
+        AppConfig: The validated application configuration object.
+
+    Raises:
+        RuntimeError: If validation fails, including details of the validation errors.
+    """
+    app_cfg = {}
 
     if filepath:
         app_cfg_file = Path(filepath)
-        fileio = app_cfg_file.open()
+        fileio = app_cfg_file.open(mode="r", encoding="utf-8")
```

Comment:
**Suggestion:** Instead of manually opening the file for reading the configuration, use a context manager to ensure the file is properly closed after loading, which is more robust and Pythonic. [best practice, importance: 8]
```suggestion
with app_cfg_file.open(mode="r", encoding="utf-8") as fileio:
    app_cfg = toml.load(fileio)
```